### ISSUE TITLE
[Feature] Add integer division to math block

### ISSUE THREAD

**Issue by laurensvalk** [pybricks/support#1503]
**Is your feature request related to a problem? Please describe.**
The Move Hub doesn't support division with `/` but it does support `//` for integer division.

**Describe the solution you'd like**
Add integer division as an option to the block dropdown.

![image](https://github.com/pybricks/support/assets/12326241/d1afcec6-c27d-46a7-9865-65dae93d118a)

**Describe alternatives you've considered**
Make `/` automatically work like `//` when floats disabled, but this is perhaps less explicit.

**Comment by DanieleBenedettelli** [comment]
Hi.
When will this be implemented? 
Also in the equation block it would be needed:
![image](https://github.com/user-attachments/assets/b26e61e9-0c4d-45f1-ad36-d2673924ac83)

Thanks

**Comment by laurensvalk** [comment]
I've been focused on the new [Learn Guide](https://pybricks.com/learn/) lately, so realistically it might take a while before we can work on app feature updates.

**Comment by andybellgbc** [comment]
It took me 2+ hours of head scratching and writing, rewriting, commenting out lines and Google searches to find out it wasn't me.

If you ignore my apparently poor decision to use the Boost hub!

At least make the blocks resolve correctly for the Boost Hub. While I see you have this an enhancement - for the Boost Hub this is a bug, especially because the error message does not provide meaningful help. If it at least had the line number.

Otherwise it's great 😃

**Comment by laurensvalk** [comment]
Thanks for your note. I suppose we could make it automatic, but then it would only work if the user has selected the Move Hub block, which normally isn't a requirement.

Another nice (but more work) way would be to do it at compile time since we know which hub is connected at that time. By having mpy-cross catch it, we could even catch it for Python users.

Maybe there already is a mpy-cross flag for no floats? After all there is one for long ints I think.

**Comment by dlech** [comment]
> Maybe there already is a mpy-cross flag for no floats?

No, the MPY bytecode would have to be parsed to look for float objects.

**Comment by laurensvalk** [comment]
Just having a `//` option on the block is probably still the most sensible (and useful even on platforms that _do_ support `/`) but while my mind is at at... here's another.

---------------

Another trivial change would be to let `MP_BINARY_OP_TRUE_DIVIDE` fall through to `MP_BINARY_OP_FLOOR_DIVIDE` on platforms without floats (i.e. movehub only):

![image](https://github.com/user-attachments/assets/f1f09b28-b39e-4357-9802-6fd2345d4836)

Then you'd get the following. Same for blocks and Python. This is relatively intuitive (decimal numbers are not supported, so division will truncate), but it isn't very Pythonic.

![image](https://github.com/user-attachments/assets/aec4c50b-c940-40ed-a89d-820f150ea511)

**Comment by laurensvalk** [comment]
🎉 

![Image](https://github.com/user-attachments/assets/ffca22d4-7667-49b6-bea3-d4a4824ea883)

**Comment by laurensvalk** [comment]
This also adds descriptions the dropdown while we are at it. This is something we can do now that we're no longer using the standard blockly block. For comparison, old version:

![Image](https://github.com/user-attachments/assets/d22d6647-ab63-4b3a-9cfd-544066a71bb4)

**Comment by BertLindeman** [comment]
Also make it easier to spot the difference between plus and divide. Nice improvement.
